File: FE-RESTAURANT/src/interfaces/AccountResponseDTO.ts
import { AccountStatus } from './enums/AccountStatus';

export interface AccountResponseDTO {
  id: number; // Sửa lại thành number để khớp với backend
  username: string;
  name: string;
  email?: string; // Thêm email
  phone: string;
  status: AccountStatus;
  createAt: string; // Sử dụng string cho LocalDateTime
  
  // Thông tin khóa tài khoản
  failedAttempts?: number;
  lockTime?: string;
  lastFailedAttempt?: string;
  
  branchId?: number; // Sửa lại thành number và optional
  branchName?: string; // Optional
  roles: string[]; // Sửa lại thành array để khớp với backend Set<String>
  avatarUrl?: string; // Thêm avatarUrl
} 

File: RESTAURANT/src/main/java/com/poly/restaurant/dtos/AccountResponseDTO.java
package com.poly.restaurant.dtos;

import com.poly.restaurant.entities.enums.AccountStatus;

import jakarta.persistence.Column;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AccountResponseDTO {
    private Long id;
    private String username;
    private String name;
    private String email;
    private String phone;
    private String avatarUrl;
    private String address;
    private Double latitude;
    private Double longitude;
    private AccountStatus status;
    private LocalDateTime createAt;
    
    // Thông tin khóa tài khoản
    private Integer failedAttempts;
    private LocalDateTime lockTime;
    private LocalDateTime lastFailedAttempt;
    
    // Thông tin chi nhánh
    private Long branchId;
    private String branchName;
    
    // Danh sách tên các vai trò
    private Set<String> roles; 
}